presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestRoles.java

```diff
@@ -544,6 +555,78 @@ public void testShowRoles()
+                failsWithMessage(format("Access Denied: Cannot select from table %s", TABLE_NAME));
+    }
+
+    @Test(groups = {HIVE_CONNECTOR, ROLES, AUTHORIZATION, PROFILE_SPECIFIC_TESTS})
+    public void testGrantSelectRevokeAll()
+            throws Exception
+    {
+        onPresto().executeQuery("CREATE ROLE role1");
+        onPresto().executeQuery("GRANT role1 TO USER alice");
+        onPresto().executeQuery(format("CREATE TABLE %s (foo BIGINT)", TABLE_NAME));
+        onPresto().executeQuery(format("GRANT SELECT ON %s TO ROLE role1", TABLE_NAME));
+        QueryAssert.assertThat(onPrestoAlice().executeQuery(format("SELECT * FROM %s", TABLE_NAME))).hasNoRows();
+        onPresto().executeQuery(format("REVOKE ALL PRIVILEGES ON %s FROM ROLE role1", TABLE_NAME));
+        QueryAssert.assertThat(() -> onPrestoAlice().executeQuery(format("SELECT * FROM %s", TABLE_NAME))).
+                failsWithMessage(format("Access Denied: Cannot select from table %s", TABLE_NAME));
+    }
+
+    @Test(groups = {HIVE_CONNECTOR, ROLES, AUTHORIZATION, PROFILE_SPECIFIC_TESTS})
+    public void testGrantToRoleRevokeFromUser()
+            throws Exception
+    {
+        onPresto().executeQuery("CREATE ROLE role1");
+        onPresto().executeQuery("GRANT role1 TO USER alice");
+        onPresto().executeQuery(format("CREATE TABLE %s (foo BIGINT)", TABLE_NAME));
+        QueryAssert.assertThat(() -> onPrestoAlice().executeQuery(format("SELECT * FROM %s", TABLE_NAME))).
+                failsWithMessage(format("Access Denied: Cannot select from table %s", TABLE_NAME));
+        onPresto().executeQuery(format("GRANT SELECT ON %s TO ROLE role1", TABLE_NAME));
+        QueryAssert.assertThat(onPrestoAlice().executeQuery(format("SELECT * FROM %s", TABLE_NAME))).hasNoRows();
+        onPresto().executeQuery(format("REVOKE SELECT ON %s FROM USER alice", TABLE_NAME));
+        QueryAssert.assertThat(onPrestoAlice().executeQuery(format("SELECT * FROM %s", TABLE_NAME))).hasNoRows();
+    }
+
+    @Test(groups = {HIVE_CONNECTOR, ROLES, AUTHORIZATION, PROFILE_SPECIFIC_TESTS})
+    public void testGrantToUserRevokeFromRole()
+            throws Exception
+    {
+        onPresto().executeQuery("CREATE ROLE role1");
+        onPresto().executeQuery("GRANT role1 TO USER alice");
+        onPresto().executeQuery(format("CREATE TABLE %s (foo BIGINT)", TABLE_NAME));
+        onPresto().executeQuery(format("GRANT SELECT ON %s TO ROLE role1", TABLE_NAME));
+        onPresto().executeQuery(format("GRANT SELECT ON %s TO USER alice", TABLE_NAME));
+        QueryAssert.assertThat(onPrestoAlice().executeQuery(format("SELECT * FROM %s", TABLE_NAME))).hasNoRows();
+        onPresto().executeQuery(format("REVOKE SELECT ON %s FROM ROLE role1", TABLE_NAME));
+        QueryAssert.assertThat(onPrestoAlice().executeQuery(format("SELECT * FROM %s", TABLE_NAME))).hasNoRows();
+    }
```

Comment:
I think what you're missing is a cleanup statement to revoke the privilege from user. Perhaps add the following statements in `cleanup()` method itself, to ensure that this won't be missed in other tests too in future:
`REVOKE ALL PRIVILEGES ON %s FROM USER alice", TABLE_NAME`
`REVOKE ALL PRIVILEGES ON %s FROM PUBLIC", TABLE_NAME`